pkg/diff/diff.go

```diff
@@ -121,35 +126,34 @@ func (d Diff) updateType(scope declared.Scope, syncName string) Operation {
 	// nomos vet error. Note that as-is, it is valid to declare something owned by
 	// another object, possible causing (and being surfaced as) a resource fight.
 	canManage := CanManage(scope, syncName, d.Actual, admissionv1.Update)
+	ignoreMutation := d.Actual.GetAnnotations()[metadata.LifecycleMutationAnnotation] == metadata.IgnoreMutation
+
+	if d.sameCSMetadata() {
+		// canManage must be true when CS metadata is the same between
+		// declared and actual, so no need to check canManage
+
+		if differ.ManagementDisabled(d.Declared) {
+			return Abandon
+		}
+
+		if ignoreMutation {
+			return NoOp // ignore mutation
+		}
+
+		return Update
+	}
+
 	switch {
 	case differ.ManagementEnabled(d.Declared) && canManage:
-		if d.Actual.GetAnnotations()[metadata.LifecycleMutationAnnotation] == metadata.IgnoreMutation &&
-			d.Declared.GetAnnotations()[metadata.LifecycleMutationAnnotation] == metadata.IgnoreMutation {
-			// The declared and actual object both have the lifecycle mutation
-			// annotation set to ignore, so we should take no action as the user does
-			// not want us to make changes to the object.
-			//
-			// If the annotation is on the actual object but not the one declared in
-			// the repository, the update, which uses SSA, would not remove the annotation
-			// from the actual object. However, Config Sync would not respect the annotation
-			// on the actual object since the annotation is not declared in the git repository.
-			//
-			// If the annotation is on the declared object but not the actual one
-			// on the cluster, we need to add it to the one in the cluster.
-			return NoOp
+		if ignoreMutation {
+			return UpdateCSMetadata
 		}
 		return Update
 	case differ.ManagementEnabled(d.Declared) && !canManage:
 		// This reconciler can't manage this object but is erroneously being told to.
 		return ManagementConflict
 	case differ.ManagementDisabled(d.Declared) && canManage:
-		if metadata.HasConfigSyncMetadata(d.Actual) {
-			manager := d.Actual.GetAnnotations()[metadata.ResourceManagerKey]
-			if manager == "" || manager == declared.ResourceManager(scope, syncName) {
-				return Abandon
-			}
-		}
-		return NoOp
+		return Abandon
```

Comment:
I think the old behavior makes more sense.
- If the actual object doesn't have any CS metadata, we can return a no-op directly. 
- If the actual object has CS metadata, but its manager is different from the current reconciler, we should also return a no-op. Abandon will remove CS metadata, which might get reverted by other manager.

Let's keep the old behavior here and I'll update the design doc accordingly.

Reply:
Ok I changed it back to the old behaviour. I also created a separate PR for the Remediator changes #1486